wordpress-mobile/WordPress-iOS
# Fixes WKHTTPCookieStore Bug

Fixes an issue where WKHTTPCookieStore `getAllCookies` fetch would always timeout because of the blocked main thread. This change unblocks the main thread by dispatching to the `userInitiated` background thread and adds a `WKCookieJarTests` which does catch the issue with the previous code.

To test:

- Attempt to load a preview blog post or site page
- Notice that the page is not authenticated with cookies and `Time out waiting for WKHTTPCookieStore to get cookies` is displayed in the console.

After the change this log message is gone and the page is properly displayed when authenticated.

PR submission checklist:
